components/eamxx/src/share/grid/remap/coarsening_remapper.cpp

```diff
@@ -291,6 +304,33 @@ do_bind_field (const int ifield, const field_type& src, const field_type& tgt)
 
 void CoarseningRemapper::do_registration_ends ()
 {
+  // Before finishing the registration we need to also register the set of masks, if needed.
+  if (m_track_mask) {
+    for (int idx=0; idx<m_mask_fields_src.size(); ++idx) {
+      auto f   = m_mask_fields_src[idx];
+      auto fid = f.get_header().get_identifier();
+      register_field_from_src(fid);
+      m_mask_map_tgt.emplace(idx,m_num_registered_fields-1);
```

Comment:
doesn't `register_field_from_src` add 1 to the m_num_registered_fields tally?  I think this will always grab the last field in the map.  There may be a better way to do that though.

Reply:
Ah, yes, you're right. Can you add a 5 words comment about that? I can see myself being confused by this in the future...